src/lib/user-ip-address.test.js

```diff
@@ -0,0 +1,65 @@
+const userIpAddress = require("./user-ip-address");
+
+describe("user ip address", () => {
+  let req;
+
+  beforeEach(() => {
+    const setup = setupDefaultMocks();
+    req = setup.req;
+  });
+
+  context("without ip header", () => {
+    it("should return null when passing req but with req.headers as null", () => {
+      req.headers["forwarded"] = null;
+      const ipAddress = userIpAddress(req.headers["forwarded"]);
```

Comment:
Just realised - all these tests can be simplified.

we dont need the request object to test the `userIpAddress` function.
all the tests can be simplified to the following.

```javascript
const forwarded = null; // or whatever forwarded value you're using.
const ipAddress = userIpAddress(forwarded);
expect(...)
```

This would probably make the tests look even simpler. 

🥘 for 💭